bitwarden_license/src/app/organizations/manage/sso.component.ts

```diff
@@ -2,88 +2,132 @@ import {
 import { UserService } from 'jslib-common/abstractions/user.service';
+import { SsoConfigApi } from 'jslib-common/models/api/ssoConfigApi';
 
 import { Organization } from 'jslib-common/models/domain/organization';
 
 import { OrganizationSsoRequest } from 'jslib-common/models/request/organization/organizationSsoRequest';
+import {
+    OrganizationSsoResponse,
+    SsoUrls
+} from 'jslib-common/models/response/organization/organizationSsoResponse';
+
+import {
+    dirtyRequired,
+    dirtyRequiredIf,
+} from 'jslib-angular/validators/dirty.validator';
+
+import {
+    OpenIdConnectRedirectBehavior,
+    Saml2BindingType,
+    Saml2NameIdFormat,
+    Saml2SigningBehavior,
+    SsoType,
+} from 'jslib-common/enums/ssoEnums';
+import { SsoConfigView } from 'jslib-common/models/view/ssoConfigView';
+
+const defaultSigningAlgorithm = 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256';
 
 @Component({
     selector: 'app-org-manage-sso',
     templateUrl: 'sso.component.html',
 })
 export class SsoComponent implements OnInit {
-
-    samlSigningAlgorithms = [
-        'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256',
-        'http://www.w3.org/2000/09/xmldsig#rsa-sha384',
-        'http://www.w3.org/2000/09/xmldsig#rsa-sha512',
-        'http://www.w3.org/2000/09/xmldsig#rsa-sha1',
+    readonly ssoTypeOptions = [
+        { name: this.i18nService.t('selectType'), value: SsoType.None, disabled: true },
+        { name: 'OpenID Connect', value: SsoType.OpenIdConnect },
+        { name: 'SAML 2.0', value: SsoType.Saml2 },
     ];
 
+    readonly ssoType = SsoType;
+
     loading = true;
+    haveTestedKeyConnector = false;
     organizationId: string;
     organization: Organization;
     formPromise: Promise<any>;
 
-    callbackPath: string;
-    signedOutCallbackPath: string;
-    spEntityId: string;
-    spMetadataUrl: string;
-    spAcsUrl: string;
+    ssoUrls: SsoUrls;
 
     enabled = this.fb.control(false);
-    data = this.fb.group({
-        configType: [],
-
-        keyConnectorEnabled: [],
-        keyConnectorUrl: [],
 
-        // OpenId
-        authority: [],
-        clientId: [],
-        clientSecret: [],
+    openIdForm = this.fb.group({
+        authority: ['', dirtyRequired()],
+        clientId: ['', dirtyRequired()],
+        clientSecret: ['', dirtyRequired()],
         metadataAddress: [],
-        redirectBehavior: [],
+        redirectBehavior: [OpenIdConnectRedirectBehavior.RedirectGet],
         getClaimsFromUserInfoEndpoint: [],
         additionalScopes: [],
         additionalUserIdClaimTypes: [],
         additionalEmailClaimTypes: [],
         additionalNameClaimTypes: [],
         acrValues: [],
         expectedReturnAcrValue: [],
+    }, {
+        updateOn: 'blur',
+    });
 
-        // SAML
-        spNameIdFormat: [],
-        spOutboundSigningAlgorithm: [],
-        spSigningBehavior: [],
-        spMinIncomingSigningAlgorithm: [],
+    samlForm = this.fb.group({
```

Comment:
If at all possible, can you rename `fb` to `formBuidler` while you're in here? 
One of the tech debt items I picked up to to do this renaming :)

Reply:
But that's like... 9 more characters I have to type! Yeah that's fine :P